e2e2/test/cases/netpol/main_test.go

```diff
@@ -0,0 +1,333 @@
+package netpol
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	fwext "github.com/aws/aws-k8s-tester/e2e2/internal/framework_extensions"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/e2e-framework/klient"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+)
+
+var testenv env.Environment
+
+func TestMain(m *testing.M) {
+	cfg, err := envconf.NewFromFlags()
+	if err != nil {
+		log.Fatalf("failed to initialize test environment: %v", err)
+	}
+	testenv = env.NewWithConfig(cfg)
+
+	testenv.Setup(
+
+		func(ctx context.Context, config *envconf.Config) (context.Context, error) {
+			client, err := config.NewClient()
+			if err != nil {
+				return ctx, err
+			}
+
+			namespaces := []string{"a", "b", "c"}
+
+			servers := map[string]string{
+				"a": "a-server",
+				"b": "b-server",
+				"c": "c-server",
+			}
+
+			// 1. Install Latest CNI version
+			err = installLatestCNI(config)
+			if err != nil {
+				log.Print("Failed to install latest aws-vpc-cni on cluster")
+				return ctx, err
+			}
+
+			// 2. Create three namespaces
+			for _, ns := range namespaces {
+				err = createNamespace(ns, client, ctx)
+				if err != nil {
+					log.Printf("Failed to create namespace %s", ns)
+					return ctx, err
+				}
+			}
+
+			// 3. Create deployment and service
+			for ns, server := range servers {
+				err = createServerAndService(ns, server, 1, client, ctx)
+				if err != nil {
+					log.Printf("Failed to create deployment and service for %s", server)
+					return ctx, err
+				}
+			}
+
+			return ctx, nil
+		},
```

Comment:
Can you add waiter(s) here so that we don't try to run the test cases until your pre-reqs are met?

Reply:
The deployment function here uses wait for the resource to be ready
https://github.com/aws/aws-k8s-tester/pull/395/files#diff-4082a58579cf8719172663a0888661788612744805944b7e28a64de150aec95dR226